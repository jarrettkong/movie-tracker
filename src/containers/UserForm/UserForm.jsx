import React, { Component } from 'react';
import { login, logout } from '../../actions';
import { connect } from 'react-redux';

class Login extends Component {
	state = {
		name: '',
		email: '',
		password: '',
		error: ''
	};

	handleChange = e => {
		const { name, value } = e.target;
		this.setState({ [name]: value });
	};

	handleSubmit = (e, name) => {
		e.preventDefault();
		const { email, password } = this.state;
		let body, dir;

		if (this.props.type === 'Log In') {
			body = { email, password };
		} else {
      body = { name, email, password };
      dir = 'new';
		}
		this.fetchUser(body, dir);
	};

	fetchUser = async (body, dir) => {
		try {
			const res = await fetch(`http://localhost:3000/api/users/${dir}`, {
				method: 'POST',
				headers: { 'Content-Type': 'application/json' },
				body: JSON.stringify(body)
			});
			const json = await res.json();
			const user = json.data;
			this.props.login(user);
		} catch (err) {
			console.log(err.status, err.message);
		}
	};

  render() {
    const name = this.props.type !== 'Log In' &&
      <>
        <label htmlFor="user-name" />
        <input
          className="detail-input"
          name="name"
          placeholder="name"
          value={this.state.name}
          id="user-name"
          onChange={this.handleChange}
        />
      </>
    
		return (
			<div className="popup">
        <div className="dialog">
          <i className='fas fa-times popup-close-btn' onClick={this.props.closeUserForm}></i>
          <h3 className="dialog-header">{this.props.type}</h3>
          <form className="user-form" onSubmit={this.handleSubmit}>
            <h4 className="form-header">Enter your details:</h4>
						<label htmlFor="user-email" />
            {name}
            <input
              className="detail-input"
							type="email"
							name="email"
							placeholder="email"
							value={this.state.email}
							id="user-email"
							onChange={this.handleChange}
						/>
						<label htmlFor="user-password" />
						<input
              className="detail-input"
							type="password"
							name="password"
							placeholder="password"
							value={this.state.password}
							id="user-password"
							onChange={this.handleChange}
						/>
						<input className="login-form-btn" type="submit" value={this.props.type} />
					</form>
				</div>
			</div>
		);
	}
}

export const mapStateToProps = state => ({ user: state.user });

export const mapDispatchToProps = dispatch => {
	return {
		login: user => dispatch(login(user)),
		logout: () => dispatch(logout())
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
