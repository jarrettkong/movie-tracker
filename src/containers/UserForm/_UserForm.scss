@import '../../variables.scss';

.user-form {
  background: white;
  display: flex;
  flex-direction: column;
  padding: 10px 30px 25px 30px;

  .form-header {
    margin: 0;
    margin-left: 10px;
    text-align: left;
    font-weight: 400;
    font-size: 14px;
  }

  .detail-input,
  .login-form-btn {
    margin: 10px;
    height: 35px;
    width: 250px;
    border-radius: 5px;
  }

  .detail-input {
    padding-left: 5px;
  }

  .login-form-btn {
    border: none;
    background: rgb(26, 26, 80);
    color: white;
    font-size: 16px;

    &:hover {
      background: rgb(40, 100, 120);;
    }
  }
}

.popup {
  position: fixed;
  top: 0px;
  left: 0px;
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  background: rgba(0, 0, 0, 0.7);
  z-index: 3;
}

.dialog {
  padding: 25px 15px 30px 15px;
  background: white;
  border-radius: 5px;
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;

  .dialog-header {
    margin: 0;
    margin-bottom: 20px;
    font-size: 26px;
  }
}

.popup-close-btn {
  position: absolute;
  top: 15px;
  right: 15px;
  z-index: 3;
  color: darkred;
  font-size: 25px;
  cursor: pointer;

  &:hover {
    color: red;
  }
}

.error {
  width: 250px;
  font-size: 14px;
  line-height: 1.2;
  color: darkred;
  position: absolute;
  bottom: 5px;
}

.login-form-btn:disabled {
  background: rgb(176, 176, 176);

  &:hover {
    background: rgb(176, 176, 176);
  }
}

