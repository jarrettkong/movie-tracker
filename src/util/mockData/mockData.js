export const mockMovie = {
	adult: false,
	backdrop_path: '/7RyHsO4yDXtBv1zUU3mTpHeQ0d5.jpg',
	belongs_to_collection: {
		id: 86311,
		name: 'The Avengers Collection',
		poster_path: '/qJawKUQcIBha507UahUlX0keOT7.jpg',
		backdrop_path: '/zuW6fOiusv4X9nnW3paHGfXcSll.jpg'
	},
	budget: 356000000,
	genres: [
		{
			id: 12,
			name: 'Adventure'
		},
		{
			id: 878,
			name: 'Science Fiction'
		},
		{
			id: 28,
			name: 'Action'
		}
	],
	homepage: 'https://www.marvel.com/movies/avengers-endgame',
	id: 299534,
	imdb_id: 'tt4154796',
	original_language: 'en',
	original_title: 'Avengers: Endgame',
	overview:
		"After the devastating events of Avengers: Infinity War, the universe is in ruins due to the efforts of the Mad Titan, Thanos. With the help of remaining allies, the Avengers must assemble once more in order to undo Thanos' actions and restore order to the universe once and for all, no matter what consequences may be in store.",
	popularity: 335.962,
	poster_path: '/or06FN3Dka5tukK1e9sl16pB3iy.jpg',
	production_companies: [
		{
			id: 420,
			logo_path: '/hUzeosd33nzE5MCNsZxCGEKTXaQ.png',
			name: 'Marvel Studios',
			origin_country: 'US'
		}
	],
	production_countries: [
		{
			iso_3166_1: 'US',
			name: 'United States of America'
		}
	],
	release_date: '2019-04-24',
	revenue: 2489617092,
	runtime: 181,
	spoken_languages: [
		{
			iso_639_1: 'en',
			name: 'English'
		}
	],
	status: 'Released',
	tagline: 'Part of the journey is the end.',
	title: 'Avengers: Endgame',
	video: false,
	vote_average: 8.5,
	vote_count: 5198
};

export const mockUser = {
	id: 1,
	name: 'Bob',
	email: 'bob@bob.com',
	password: 'password'
};

export const mockPost = {
	method: 'POST',
	headers: { 'Content-Type': 'application/json' },
	body: JSON.stringify({ email: mockUser.email, password: mockUser.password })
};

export const mockPath = 'popular';

export const mockMovies = [
	{
		id: 456740,
		poster: 'https://image.tmdb.org/t/p/w300/bk8LyaMqUtaQ9hUShuvFznQYQKR.jpg',
		title: 'Hellboy'
	},
	{
		id: 447404,
		poster: 'https://image.tmdb.org/t/p/w300/wgQ7APnFpf1TuviKHXeEe3KnsTV.jpg',
		title: 'Pok√©mon Detective Pikachu'
	}
];

export const mockState = {
	trendingMovies: mockMovies,
	topRatedMovies: mockMovies,
	nowPlayingMovies: mockMovies,
	upcomingMovies: mockMovies,
	user: mockUser
};
