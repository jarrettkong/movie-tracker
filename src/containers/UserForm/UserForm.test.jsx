import React from 'react';
import { UserForm } from './UserForm';
import Enzyme, { shallow } from 'enzyme';
import { mockMovies } from '../../util/mockData/mockData';

describe('UserForm', () => {
  let wrapper;

  it('Should match the snapshot', () => {
    wrapper = shallow(<UserForm type="Log In" />)

    expect(wrapper).toMatchSnapshot();
  });

  it('Should match the snapshot', () => {
    wrapper = shallow(<UserForm type="Sign Up" />)

    expect(wrapper).toMatchSnapshot();
  });

  //simulate change then simulate submit
  it('Should match the snapshot', () => {
    wrapper = shallow(<UserForm type="Sign Up" />)

    expect(wrapper).toMatchSnapshot();
  });

})
