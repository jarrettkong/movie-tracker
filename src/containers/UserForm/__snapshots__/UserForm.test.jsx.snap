// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm Should match the snapshot as a log in form 1`] = `ShallowWrapper {}`;

exports[`UserForm Should match the snapshot as a sign up form 1`] = `ShallowWrapper {}`;
